includes/class-wc-payments-order-service.php

```diff
@@ -875,6 +882,28 @@ private function mark_order_held_for_review_for_fraud( $order, $intent_data ) {
 		$this->set_intention_status_for_order( $order, $intent_data['intent_status'] );
 	}
 
+	/**
+	 * Given the charge, adds the application_fee_amount from the charge to the given order as metadata.
+	 *
+	 * @param WC_Order               $order The order to update.
+	 * @param WC_Payments_API_Charge $charge The charge to get the application_fee_amount from.
+	 */
+	public function attach_transaction_fee_to_order( $order, $charge ) {
+		try {
+			if ( $charge && null !== $charge->get_application_fee_amount() ) {
```

Comment:
IMO, we should declare param types here, and then force consumers to check these param types before passing them to use this method. 

With this, we do not need to check `$charge` inside this method (currently, consumers can actually pass `null`) and give the right expectation for consumers. 

Not a blocker but it would reduce the cognitive load in using this method in the future. 

```suggestion
	public function attach_transaction_fee_to_order( WC_Order $order, WC_Payments_API_Charge $charge ) {
		try {
			if ( null !== $charge->get_application_fee_amount() ) {
```

Reply:
> With this, we do not need to check `$charge` inside this method (currently, consumers can actually pass `null`) 
> Not a blocker but it would reduce the cognitive load in using this method in the future.

There are few flows which have (validly) $charge as null at some points, I would rather have the cognitive load within the function than its consumers (multiple points of the payment processing functions), since the code there are already quite nested and difficult to parse.